docker/models/images.py

```diff
@@ -0,0 +1,94 @@
+import re
+
+import six
+
+from ..errors import BuildError
+from ..utils.json_stream import json_stream
+from .resource import Collection, Model
+
+
+class Image(Model):
+    def __repr__(self):
+        return "<%s: %s>" % (self.__class__.__name__, self.name)
+
+    @property
+    def name(self):
+        return self.tags[0] if self.tags else self.short_id
+
+    @property
+    def short_id(self):
+        """
+        Returns the "sha256:" prefix plus 10 characters of the ID
+        """
+        if self.id.startswith('sha256:'):
+            return self.id[:17]
+        return self.id[:10]
+
+    @property
+    def tags(self):
+        return [
+            tag for tag in self.attrs.get('RepoTags', [])
+            if tag != '<none>:<none>'
+        ]
+
+    def history(self):
+        return self.client.api.history(self.id)
+
+    def push(self):
```

Comment:
I don't think taking an object-oriented (and inheritance-based) approach here is harmful at all.